pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1156,6 +1163,33 @@ func (self *awsDisk) describeVolume() (*ec2.Volume, error) {
 	return volumes[0], nil
 }
 
+// getAttachmentStatus returns current attachment status of the volume
+func (self *awsDisk) getAttachmentStatus() (string, error) {
+	info, err := self.describeVolume()
+	if err != nil {
+		return "", err
+	}
+	if len(info.Attachments) > 1 {
+		glog.Warningf("Found multiple attachments for volume: %v", info)
+	}
+	attachmentStatus := ""
+	for _, attachment := range info.Attachments {
+		if attachmentStatus != "" {
+			glog.Warning("Found multiple attachments: ", info)
```

Comment:
Yeah it is probably superfluous with the len(info.Attachments) check, but EBS seems so flaky at times that I wanted to program really defensively.  This will only fire if somehow a volume got attached twice (which _should_ be impossible)